client/src/components/DeleteWorkspaceDialog.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+
+  <v-dialog
+    v-model="dialog"
+    width="700"
+    >
+
+    <template v-slot:activator="{ on }">
+      <v-tooltip right>
+        <template v-slot:activator="tooltip">
+          <v-scroll-x-transition>
+            <v-btn
+              icon
+              small
+              text
+              v-if="somethingChecked"
+              v-on="on"
+              >
+              <v-icon color="red accent-3" size="22px">delete_sweep</v-icon>
+            </v-btn>
+          </v-scroll-x-transition>
+        </template>
+        <span>Delete selected</span>
+      </v-tooltip>
+    </template>
+
+    <v-card>
+      <v-card-title
+        class="headline pb-0 pt-3"
+        primary-title
+        >
+        Delete Workspaces
+      </v-card-title>
+
+      <v-card-text class="px-4 pt-4 pb-1">
+        You are about to delete {{ selection.length }} workspace{{plural}}. <strong>Are you sure?</strong>
+      </v-card-text>
+
+      <v-divider />
+
+      <v-card-actions class="px-4 py-3">
+        <v-spacer />
+        <v-btn
+          depressed
+          color="error"
+          @click="execute"
+          :disabled="disabled"
```

Comment:
For some reason the button looks disabled initially, but after a couple seconds turns red. Is this delay intentional?

Reply:
It was:
> One issue is that the GET /workspaces endpoint doesn't seem to immediately update after the DELETE /workspaces/<id> call succeeds. To get around this, I have a one-second delay before the Sidebar component refreshes the workspaces list. One second works but is not guaranteed formally.

Reply:
Yeah, it's my way of slowing the user down before they do something irreversible.

It's quite likely bad UX. The better solution is to not actually delete, but simply "archive" the workspaces so they no longer show up, but are not actually deleted. See #214.